src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -316,4 +694,223 @@ private AIState startBuilding()
                 return AIState.COMPLETE_BUILD;
         }
     }
+
+    private boolean handleMaterials(@NotNull final Block block, @NotNull final IBlockState blockState)
+    {
+        //Breaking blocks doesn't require taking materials from the citizens inventory
+        if (block == Blocks.AIR)
+        {
+            return true;
+        }
+
+        if (isBlockFree(block, block.getMetaFromState(blockState)))
+        {
+            return true;
+        }
+
+        return !checkOrRequestItems(BlockUtils.getItemStackFromBlockState(blockState));
+    }
+
+    /**
+     * Defines blocks that can be built for free.
+     *
+     * @param block    The block to check if it is free.
+     * @param metadata The metadata of the block.
+     * @return true or false.
+     */
+    public static boolean isBlockFree(@Nullable final Block block, final int metadata)
+    {
+        return block == null
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (8) used in the expression (maximum allowed 4). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1067)